### PR TITLE
Update documentation on RuntimeMethodHandle.GetFunctionPointer

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Update the documentation on `RuntimeMethodHandle.GetFunctionPointer` now that we have function pointers in C# based on the discussion in https://github.com/dotnet/runtime/issues/65853.

### CODE REVIEW

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: xml/System/RuntimeMethodHandle.xml [hunk 1]
```diff
@@ -251,7 +251,7 @@
...
 ## Remarks  
-   The returned <xref:System.IntPtr> is meant for internal runtime usage and should not be used to call the function from native code. Use <xref:System.Runtime.InteropServices.Marshal.GetFunctionPointerForDelegate%2A?displayProperty=nameWithType> in order to get a function pointer that can be passed to native code.  
+   If the method this handle represents is static, then the returned value can be cast to a C# managed function pointer type with the same signature. If the method has the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute%2A?displayProperty=nameWithType> attribute, then the returned value is an unmanaged function pointer with the same calling convention as specified in the attribute. For all other cases, the value is not easily usable from user code and is meant exclusively for usage within the runtime. For methods that do not have the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute%2A?displayProperty=nameWithType>, use <xref:System.Runtime.InteropServices.Marshal.GetFunctionPointerForDelegate%2A?displayProperty=nameWithType> in order to get a function pointer that can be passed to native code.
```
Comment (gewarren) [inline xml/System/RuntimeMethodHandle.xml:254]: Is "for all other cases" different to "for methods that don't have the UnmanagedCallersOnlyAttribute"? If so, then I think the wording needs to be clarified for what "all other cases" refers to.

File: xml/System/RuntimeMethodHandle.xml [hunk 2]
```diff
@@ -251,7 +251,7 @@
...
 ## Remarks  
-   The returned <xref:System.IntPtr> is meant for internal runtime usage and should not be used to call the function from native code. Use <xref:System.Runtime.InteropServices.Marshal.GetFunctionPointerForDelegate%2A?displayProperty=nameWithType> in order to get a function pointer that can be passed to native code.  
+   If the method has the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute%2A?displayProperty=nameWithType> attribute, then the returned value is an unmanaged function pointer with the same calling convention as specified in the attribute. If the method this handle represents is static, then the returned value can be cast to a C# managed function pointer type with the same signature. For instance method handles, the value is not easily usable from user code and is meant exclusively for usage within the runtime. For methods that do not have the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute%2A?displayProperty=nameWithType>, use <xref:System.Runtime.InteropServices.Marshal.GetFunctionPointerForDelegate%2A?displayProperty=nameWithType> in order to get a function pointer that can be passed to native code.
```
Comment (gewarren) [inline xml/System/RuntimeMethodHandle.xml:254]:
```suggestion
 
 If the method has the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute%2A?displayProperty=nameWithType> attribute, then the returned value is an unmanaged function pointer with the same calling convention as specified in the attribute. If the method this handle represents is static, then the returned value can be cast to a C# managed function pointer type with the same signature. For instance method handles, the value is not easily usable from user code and is meant exclusively for usage within the runtime.
   
 For methods that do not have the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute%2A?displayProperty=nameWithType>, use <xref:System.Runtime.InteropServices.Marshal.GetFunctionPointerForDelegate%2A?displayProperty=nameWithType> in order to get a function pointer that can be passed to native code.  
```

**Review by gewarren — APPROVED** [review]
I broke it out into two paragraphs - but also fine as-is if you don't like the suggestion.

File: xml/System/RuntimeMethodHandle.xml [hunk 3]
```diff
@@ -251,7 +251,10 @@
         <remarks>
           <format type="text/markdown"><![CDATA[  
 ## Remarks  
-   The returned <xref:System.IntPtr> is meant for internal runtime usage and should not be used to call the function from native code. Use <xref:System.Runtime.InteropServices.Marshal.GetFunctionPointerForDelegate%2A?displayProperty=nameWithType> in order to get a function pointer that can be passed to native code.  
+ 
+ If the method has the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute%2A?displayProperty=nameWithType> attribute, then the returned value is an unmanaged function pointer with the same calling convention as specified in the attribute. If the method this handle represents is static, then the returned value can be cast to a C# managed function pointer type with the same signature. For instance method handles, the value is not easily usable from user code and is meant exclusively for usage within the runtime.
```
Comment (gewarren) [inline xml/System/RuntimeMethodHandle.xml:255]:
```suggestion
 If the method has the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute?displayProperty=nameWithType> attribute, then the returned value is an unmanaged function pointer with the same calling convention as specified in the attribute. If the method this handle represents is static, then the returned value can be cast to a C# managed function pointer type with the same signature. For instance method handles, the value is not easily usable from user code and is meant exclusively for usage within the runtime.
```

File: xml/System/RuntimeMethodHandle.xml [hunk 4]
```diff
@@ -251,7 +251,10 @@
...
+   
+ For methods that do not have the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute%2A?displayProperty=nameWithType>, use <xref:System.Runtime.InteropServices.Marshal.GetFunctionPointerForDelegate%2A?displayProperty=nameWithType> in order to get a function pointer that can be passed to native code.
```
Comment (gewarren) [inline xml/System/RuntimeMethodHandle.xml:257]:
```suggestion
 For methods that don't have the <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute?displayProperty=nameWithType>, use <xref:System.Runtime.InteropServices.Marshal.GetFunctionPointerForDelegate%2A?displayProperty=nameWithType> to get a function pointer that can be passed to native code.  
```

**Review by gewarren — COMMENTED** [review]
Fixed the xrefs.

**Pull request merged**